src/common/connection.js

```diff
@@ -176,7 +176,8 @@ class Connection extends EventEmitter {
         this._ws.on('error', error =>
           this.emit('error', 'websocket', error.messsage, error));
         this._ws.on('message', this._onMessage.bind(this));
-        this._onUnexpectedCloseBound = this._onUnexpectedClose.bind(this);
+        this._onUnexpectedCloseBound = this._onUnexpectedClose.bind(this,
+          resolve, reject);
```

Comment:
Hm. This is one is simple - after creating new `RippleAPI` object and calling `connect`, `close` event can came before `open`. Maybe it is not according to specification, but it happens in real life :smile: 
So code just checks for that situation.